src/ServiceStack.OrmLite/Expressions/SqlExpression.cs

```diff
@@ -682,6 +683,12 @@ private SqlExpression<T> OrderByInternal(Expression keySelector)
             return this;
         }
 
+        private static bool ObjectIsClass(object obj)
+        {
+            return obj != null &&
+                   !(obj.GetType().IsPrimitive() || obj.GetType().IsValueType() || obj is string);
+        }
+
```

Comment:
Why isn't this using `ob.GetType().IsClass()`?

Reply:
I need to determine value types include ``string``. The ``ob.GetType().IsClass()`` is ``true`` for ``string``. Also I've found solution for this problem in your project in method ``ServiceStack.OrmLite.OrmLiteSPStatement.ConvertToScalarList<T>()``:
```
public List<T> ConvertToScalarList<T>()
        {
            if (!((typeof(T).IsPrimitive()) || typeof(T).IsValueType() || (typeof(T) == typeof(string)) || (typeof(T) == typeof(String))))
                throw new Exception("Type " + typeof(T).Name + " is a non primitive type. Use ConvertToList function.");
...
        }
```
So I can change like this:
```
        private static bool ObjectIsClass(object obj)
        {
            return obj != null &&
                   !(obj is string) &&
                   obj.GetType().IsClass();
        }
```
Is it necessary to do so?

Reply:
IMO It makes more sense for the method called `ObjectIsClass()` to use the `IsClass()` API then using inverted checks of IsPrimitive + IsValueType.

But I'm not thrilled of the name of the method as it doesn't provide any indication of what it's guarding against, what other types is this method for apart from `PartialSqlString` ?

Reply:
One more type is ``SelectList`` (``VisitNew(NewExpression nex``). So i've decided exclude all reference types.

Reply:
It's more descriptive to list the concrete types so we know when this code-path is used.

Reply:
Something like this?
```
        private static bool ObjectIsSqlClass(object obj)
        {
            return obj != null &&
                   (obj is PartialSqlString ||
                    obj is SelectList);
        }
```

Reply:
Just IsSqlClass is fine.